src/runtime/storage/pouchdb/pouch-db-collection.ts

```diff
@@ -125,7 +125,9 @@ export class PouchDbCollection extends PouchDbStorageProvider implements Collect
 
       const retrieveItem = async item => {
         const ref = item.value;
-        return {id: ref.id, value: await this.backingStore.get(ref.id), keys: item.keys};
+        const backedValue = await this.backingStore.get(ref.id);
```

Comment:
Maybe the assert should be at `backingStore.get`? I haven't studied this code, so whatever you think is best, but that you had to do this twice is a signal IMO.

Reply:
Revised to handle asserts entirely within `PouchDbCollection.get`. Pouch now behaves differently from Firebase in this respect but all of this code is being refactored by @shans and in the interim this could help us debug Pouch issues.